core/domain/beam_job_services_test.py

```diff
@@ -0,0 +1,300 @@
+                the Cloud Dataflow service; callers cannot mutate it.
+            current_state_time: str. The timestamp associated with the current
+                state. A timestamp in RFC3339 UTC "Zulu" format, with nanosecond
+                resolution and up to nine fractional digits. Examples:
+                "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+
+        Returns:
+            Context manager. A context manager in which
+            subprocess.check_output() always returns a mock gcloud response.
+        """
+        value = json.dumps({
+            'currentState': current_state,
+            'currentStateTime': current_state_time,
+        })
+        return (
+            self.swap_to_always_return(subprocess, 'check_output', value=value))
+
+    def create_beam_job_run_model(
+            self, job_id=None, job_name='FooJob', job_arguments=None,
+            job_state=beam_job_models.BeamJobState.RUNNING.value):
+        """Returns a new BeamJobRunModel with convenient default values.
+
+        Args:
+            job_id: str|None. The ID of the job. If None, a value is generated.
+            job_name: str. The name of the job class that implements the
+                job's logic.
+            job_state: str. The state of the job at the time the model was last
+                updated.
+            job_arguments: list(str)|None. The arguments provided to the job
+                run. If None, an empty list will be used instead.
+
+        Returns:
+            BeamJobRunModel. The new model.
+        """
+        if job_id is None:
+            job_id = python_utils.NEXT(self._id_iter)
+        if job_arguments is None:
+            job_arguments = []
+        return beam_job_models.BeamJobRunModel(
+            id=job_id, job_name=job_name, job_arguments=job_arguments,
+            latest_job_state=job_state)
+
+    def assert_domain_equals_model(self, beam_job_run, beam_job_run_model):
+        """Asserts that the domain object has the same values as the model.
+
+        Args:
+            beam_job_run: BeamJobRun. The domain object.
+            beam_job_run_model: BeamJobRunModel. The model.
+
+        Raises:
+            AssertionError. The domain object and model have at least one
+                difference.
+        """
+        self.assertEqual(beam_job_run.job_id, beam_job_run_model.id)
+        self.assertEqual(beam_job_run.job_name, beam_job_run_model.job_name)
+        self.assertEqual(
+            beam_job_run.job_state, beam_job_run_model.latest_job_state)
+        self.assertEqual(
+            beam_job_run.job_arguments, beam_job_run_model.job_arguments)
+        self.assertEqual(
+            beam_job_run.job_started_on, beam_job_run_model.created_on)
+        self.assertEqual(
+            beam_job_run.job_updated_on, beam_job_run_model.last_updated)
+
+    def assert_domains_equal_models(self, beam_job_runs, beam_job_run_models):
```

Comment:
Don't define this and the one above -- just inline it. You only use it in one place anyway.